src/main/scala/chisel3/util/AttributeAnnotation.scala

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.utils
+
+import firrtl.AttributeAnnotation
+import firrtl.annotations.Named
+
+import chisel3.Data
+import chisel3.experimental.{annotate, requireIsAnnotatable, ChiselAnnotation}
+
+object annotateAttribute {
+  def apply[T <: Data](target: T, annoString: String): Unit = {
+    requireIsAnnotatable(target, "target must be annotatable")
+    annotate(new ChiselAnnotation {
+      def toFirrtl = AttributeAnnotation(target.toNamed, annoString)
+    })
+  }
+}
```

Comment:
This needs ScalaDoc with a simple example. I also think we should have a name that doesn't refer to annotations as that's an implementation detail that will hopefully change. Not sure what we should call it `addAttribute`? Any inspiration?

Reply:
I'll make sure to capture the list of what CIRCT can process in the docs as well.